yt/geometry/_selection_routines/region_selector.pxi

```diff
@@ -41,42 +41,23 @@ cdef class RegionSelector(SelectorObject):
                         i, region_width[i]))
 
         for i in range(3):
+            # First, we check if any criteria requires a period check,
+            # without any adjustments.  This is for short-circuiting the
+            # short-circuit of the loop down below in mask filling.
+            if LE[i] < DLE[i] or LE[i] > DRE[i] or RE[i] > DRE[i]:
+                self.check_period[i] = True
+            # shift so left_edge guaranteed in domain
+            if LE[i] < DLE[i]:
+                LE[i] += DW[i]
+                RE[i] += DW[i]
+            elif LE[i] > DRE[i]:
+                LE[i] -= DW[i]
+                RE[i] -= DW[i]
 
-            if p[i]:
-                # First, we check if any criteria requires a period check,
-                # without any adjustments.  This is for short-circuiting the
-                # short-circuit of the loop down below in mask filling.
-                if LE[i] < DLE[i] or LE[i] > DRE[i] or RE[i] > DRE[i]:
-                    self.check_period[i] = True
-                # shift so left_edge guaranteed in domain
-                if LE[i] < DLE[i]:
-                    LE[i] += DW[i]
-                    RE[i] += DW[i]
-                elif LE[i] > DRE[i]:
-                    LE[i] -= DW[i]
-                    RE[i] -= DW[i]
-            else:
-                if LE[i] < DLE[i] or RE[i] > DRE[i]:
-                    raise RuntimeError(
-                        "yt attempted to read outside the boundaries of "
-                        "a non-periodic domain along dimension %s.\n"
-                        "Region left edge = %s, Region right edge = %s\n"
-                        "Dataset left edge = %s, Dataset right edge = %s\n\n"
-                        "This commonly happens when trying to compute ghost cells "
-                        "up to the domain boundary. Two possible solutions are to "
-                        "select a smaller region that does not border domain edge "
-                        "(see https://yt-project.org/docs/analyzing/objects.html?highlight=region)\n"
-                        "or override the periodicity with\n"
-                        "ds.force_periodicity()" % \
-                        (i, dobj.left_edge[i], dobj.right_edge[i],
-                         dobj.ds.domain_left_edge[i], dobj.ds.domain_right_edge[i])
-                    )
             # Already ensured in code
             self.left_edge[i] = LE[i]
             self.right_edge[i] = RE[i]
             self.right_edge_shift[i] = RE[i] - DW[i]
-            if not self.periodicity[i]:
```

Comment:
I'm wondering particularly about these lines--`right_edge_shift` gets used below in a few places and I want to make sure that it still does the right thing in the case of non-periodic boundaries. It's particles in particular that I am worried about

Reply:
I agree (and note my comments above). This was a section of the code that I was most worried about. Let me take a closer look to make sure what I am doing makes logical sense. Again, if you have a dataset that would be a good test case, I would appreciate it!

Reply:
@mameehan5 I apologize again for the delay. Hoping to get this settled this week. It won't take long, just trying to find the time.

Reply:
No problem, just let me know when you get the chance!

Reply:
Ok--I've looked at this a bit more, and I think I'm concerned that we don't fully understand the ramifications of taking away the line that sets `self.right_edge_shift[i] = -np.inf` for a non-periodic boundary. The rest of it looks ok to me. If you had a selection box that was fully within the domain, then it looks like you would never have to worry about `right_edge_shift`, because whether it was `right_edge-domain_width` or `-np.inf` it would still be outside of the domain. 

But what if you didn't? What if you had code that defined a selection box that extended way off the domain in some direction, say over a domain width? Obviously for a non-periodic boundary it should only keep the data that's within the domain, and this is guaranteed if `right_edge_shift = -np.inf`. But if you had `right_edge_shift = right_edge - domain_width` then it seems like the behavior could be different. 

Is there any way to restore this line without breaking everything you're trying to accomplish? That would be preferable, if it's possible.

Reply:
My example is obviously a bit silly, but I can see code that iterates over box regions for whatever reason happening upon a box that straddles a boundary, and we'd want to make sure it was always "doing the right thing" from a selection perspective even if the region extending far off the domain makes little physical sense.

Reply:
That definitely makes sense! I think I should be able to restore that. Let me test it out and get back to you.